### DISCUSSION TITLE
update pyarrow requirement to ^14.0.0?

### DISCUSSION THREAD

**Discussion by emileten** [developmentseed/lonboard#227] [Ideas]
pyarrow below `14.0.0` is incompatible with python 3.12. Since lonboard specifies pyarrow versions with `^13.0.0` (`14.0.0` excluded), installing lonboard with python 3.12 leads to failures. 

Should we change the pyarrow requirements to `pyarrow=^14.0.0` to allow for python 3.12, or explicitly indicate that python 3.12 is not supported?

**Comment by kylebarron** [comment]
We needed to bump to 14.0.1 anyways because of a security disclosure: https://github.com/advisories/GHSA-5wvp-7f3h-6wmm. Fixed in https://github.com/developmentseed/lonboard/pull/230
